test/fw/ptl/lib/pbs_testlib.py

```diff
@@ -14002,6 +13989,56 @@ def add_mom_dyn_res(self, custom_resource, script_body=None,
         self.add_config(a)
         return res_file
 
+    def enable_cgroup_cset(self):
+        """
+        Configure and enable cgroups hook
+        """
+        # check if cgroups subsystems including cpusets are mounted
+        with open(os.path.join(os.sep, 'proc', 'mounts'), 'r') as fd:
+            mounts = fd.read()
+        if 'cgroup' in mounts and mounts.count('cpuset') >= 2:
+            pbs_conf_val = self.du.parse_pbs_config(self.hostname)
+            f1 = os.path.join(pbs_conf_val['PBS_EXEC'], 'lib',
+                              'python', 'altair', 'pbs_hooks',
+                              'pbs_cgroups.CF')
+            vpn1 = r'"vnode_per_numa_node"\s*:\s*false,'
+            vpn2 = '"vnode_per_numa_node"   : true,'
+            uht1 = r'"use_hyperthreads"\s*:\s*false,'
+            uht2 = '"use_hyperthreads"      : true,'
+            mem1 = r'"memory"\s*:\s*{'
+            mem2 = r'"memsw"\s*:\s*{'
+            ena1 = r'"enabled"\s*:\s*true,'
+            ena2 = '"enabled"            : false,'
+            # set vnode_per_numa_node = true, use_hyperthreads = true
+            with open(f1, "r") as cfg:
+                lines = cfg.readlines()
+            for i, line in enumerate(lines):
+                if re.search(vpn1, line):
```

Comment:
Voltaire, what happens when the search fails to find vnode_per_numa_node and use_hyperthreads in the config file? Shouldn't we be adding the feilds even in cases search fails because the lines are not present?

Reply:
Since I am getting the data from the default config file pbs_cgroups.CF, if vnode_per_numa_node and use_hyperthreads lines are missing then there is a bug in that file.  Even if the lines were intentionally left out of the pbs_cgroups.CF if we added those lines later on, then it would not be based on a default configuration anymore.

Reply:
Oh ok, I get it. But, what if in future we remove (maybe by mistake) it from the default config for some reason. The reason that you are adding it to revert_mom is that you want to provide a consistent configuration that works on cpuset machines. I feel it is better to just replace the lines (if found) or add them in case it is not there.

Reply:
I will make changes accordingly in the next iteration.

Reply:
Thanks

Reply:
@arungrover  and @vccardenas But by adding it to revert_mom, testing will now miss the fact that those settings are not there in the PBS_EXEC area.  Maybe rather than just setting it, an error should be printed if those 2 particular configurations are not found?

Reply:
I think to check that default config is proper or not isn't really the purpose of a function that's supposed to enable cgroups for cpuset machines.
I actually feel that such a check is not needed. But If it is, then it can be done from the setUpClass itself to do some sanity check on the provided configuration and then bail out if a problem exists.

Reply:
The file is a json file, so you should be able to read it, convert it to a dictionary, change the values, and print out json.